[fy17ohhh] iP

fy17ohhh opened the pull request:
# Jade
>  "One man's 'magic' is another man's engineering." - Robert A. Heinlein [(source)](https://www.accu.co.uk/p/409-engineering-quotes)

Jade is your task manager who helps you remember things you need to do. It's,

- text-based
- fast to use
- ~~EASY~~ SUPER EASY to learn

All you need to do is,

1. download the JAR file from [here](https://github.com/fy17ohhh/ip/releases/tag/A-Jar#:~:text=3-,jade.jar,-18.5%20KB).
1. double-click it
1. add your tasks
1. let it manage your tasks for you :+1: 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public static void main(String[] args) {
    Jade myJade = new Jade("data/jadeList.txt");
    myJade.run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

fy17ohhh committed:
Rename, Greet, Exit

fy17ohhh committed:
Echo

fy17ohhh committed:
Add,List

fy17ohhh committed:
Use a class to represent tasks

fy17ohhh committed:
Add the ability to mark tasks as done

fy17ohhh committed:
Use Inheritance to support multiple task types

fy17ohhh committed:
Add support for tracking three types of tasks

fy17ohhh committed:
Automated Text UI Testing

fy17ohhh committed:
update Class Name and Test Files

fy17ohhh committed:
Added custom exception class

fy17ohhh committed:
Refactored code and add new feature to handle incorrect inputs

fy17ohhh committed:
Add support for deleting tasks

fy17ohhh committed:
Use Enums for Commands

fy17ohhh committed:
Add feature to save change in harddisk

fy17ohhh committed:
Merge Level-7 to master

fy17ohhh committed:
Use LocalDate for dates in tasks

fy17ohhh committed:
Print dates in format and tasks on a specific date

fy17ohhh committed:
Merge Level-8

fy17ohhh committed:
Update ignore files and classify class files

fy17ohhh committed:
Update ignore file

fy17ohhh committed:
Update ignore file

fy17ohhh committed:
Make the code more OOP and organize into pkgs

fy17ohhh committed:
Merge OOP and pkg branch

fy17ohhh committed:
Update files

fy17ohhh committed:
Implement JUnit test for task class

fy17ohhh committed:
Automate project run and unit test using gradle

fy17ohhh committed:
Add JUnit test for Task and TaskList classes

fy17ohhh committed:
Update Task and TaskList tests

fy17ohhh committed:
Update to save to local only after exiting program

fy17ohhh committed:
Bug fix: runShadow with System.in

fy17ohhh committed:
Add JavaDoc comments to all classes

fy17ohhh committed:
Tweak the code to comply with a coding standard

fy17ohhh committed:
Add a feature to find a task by searching for a keyword

fy17ohhh committed:
Resolve merge conflic

fy17ohhh committed:
Resolve merge conflict

fy17ohhh committed:
Bug fix: method name unmatches fixed

fy17ohhh committed:
Update JavaDoc comments

fy17ohhh committed:
Fix JavaDoc typos

fy17ohhh committed:
Add checkstyle files to detect coding style violations

fy17ohhh committed:
Add Javafx files for GUI

fy17ohhh committed:
Modify Command methods

fy17ohhh committed:
Bug fix: refactor find and list command

fy17ohhh committed:
Update exit function

fy17ohhh committed:
Bug fix: save file to local

fy17ohhh committed:
Add GUI to application

fy17ohhh committed:
Use varargs for checking empty descriptions

fy17ohhh committed:
Update README

fy17ohhh committed:
Add time feature in task date

yuechen2001 commented on src/main/java/jade/commands/FindCommand.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws JadeException {
```
The names of the variables are very clear and easy to understand, good job!

yuechen2001 commented on src/main/java/jade/parser/Parser.java:
```diff
@@ -0,0 +1,88 @@
+import jade.data.Deadline;
+import jade.data.Event;
+import jade.data.Todo;
+import jade.commands.*;
```
Instead of using *, it would be better if imported classes are listed explicitly.

yuechen2001 commented on src/main/java/jade/data/TaskList.java:
```diff
@@ -0,0 +1,54 @@
+        return taskList.get(index);
+    }
+
+    public String listFormatter() {
```
Would it be clearer to the reader, if some comments (e.g. before and after formatting examples) were provided regarding the use of taskFormatter and listFormatter methods?

yuechen2001 reviewed (commented):
I think you did a great job with your project! Keep up the good work thus far :)

yuechen2001 commented on src/main/java/jade/storage/Storage.java:
```diff
@@ -0,0 +1,106 @@
+     *
+     * @return A list of user tasks.
+     * @throws JadeException If directory or file does not exit
+     */
```
I really like how clear and concise the comments on all of your methods are good job!

yuechen2001 commented on src/main/java/jade/storage/Storage.java:
```diff
@@ -0,0 +1,106 @@
+     * @return A list of user tasks.
+     * @throws JadeException If directory or file does not exit
+     */
+    public List<Task> load() throws JadeException {
```
This method felt a bit long and difficult to understand, maybe it would be better to separate it out into helper functions and the main function?

fy17ohhh committed:
Use assertions to document important assumptions

Add assertions for the default cases in switch statements in Parser and
Storage classes to indicate the assumption that one of the cases will
always be executed, i.e. user will only enter valid commands during
runtime.

An assertion for the load() method in Storage class is also added, as
the existing local file is assumed to have no empty strings.

fy17ohhh committed:
Refactor code to improve code quality

Too many magic literals used in the previous commits. Instead, use named
constants to improve the readability of code.

Add more comments to specify what the methods are supposed to do, and
the rationale for the current implementations.

Use SLAP in switch statements in Parser class, while also applying
extract method to turn the fragment into a method whose name explains
the purpose of the method.

fy17ohhh committed:
Replace loops with java streams

fy17ohhh committed:
Merge pull request #2 from fy17ohhh/branch-A-CodeQuality

Refactor code to improve code quality

fy17ohhh committed:
Fix conficts

fy17ohhh committed:
Fix confict

fy17ohhh committed:
Merge pull request #3 from fy17ohhh/branch-A-Streams

Replace loops with java streams

fy17ohhh committed:
Merge branch 'master' into branch-A-Assertions

fy17ohhh committed:
Merge pull request #1 from fy17ohhh/branch-A-Assertions

Use assertions to document important assumptions

fy17ohhh committed:
Fix bug: assertion branch merging error

fy17ohhh committed:
Set up Continuous Integration

fy17ohhh committed:
Delete unused imports

fy17ohhh committed:
Delete I/O redirection tests

fy17ohhh committed:
Add help page for using commands

fy17ohhh committed:
Provide support for managing recurring tasks

Dexter-Wong commented on src/main/java/jade/commands/AddCommand.java:
```diff
@@ -0,0 +1,41 @@
+package jade.commands;
+
+import jade.data.Task;
```
Nice spacing of imports

Dexter-Wong commented on src/main/java/jade/commands/InvalidCommand.java:
```diff
@@ -0,0 +1,35 @@
+        this.invalidMessage = exceptionMessage;
+    }
+    /**
+     * @inheritDoc This implementation prints an error message to user
```
Good use of inheritdoc

Dexter-Wong commented on src/main/java/jade/data/Task.java:
```diff
@@ -0,0 +1,81 @@
+        return String.format("T | %s | %s\n", statusFormatter(), description);
+    }
+
+    public boolean containsKeyword(String keyword) {
```
perhaps make a more boolean-obvious name

Dexter-Wong commented on src/main/java/jade/commands/FindCommand.java:
```diff
@@ -0,0 +1,53 @@
+     * @inheritDoc This implementation prints all tasks that contain the keyword.
+     */
+    @Override
+    public String execute(TaskList taskList, Storage storage) throws JadeException {
```
nice formatting

Dexter-Wong commented on src/main/java/jade/gui/Main.java:
```diff
@@ -0,0 +1,32 @@
+/**
+ * A GUI for Jade using FXML.
+ */
+public class Main extends Application {
```
Good use of javadocs

Dexter-Wong reviewed (commented):
LGTM, nice job.

fy17ohhh committed:
Bug fix: parsing date format error

fy17ohhh committed:
Add extension of help page and recurring task support

fy17ohhh committed:
Improve code quality: comments format fixed

fy17ohhh committed:
Allow resizing of the Window

fy17ohhh committed:
Improve the GUI

fy17ohhh committed:
Add personality: change feedback messages

fy17ohhh committed:
Add more personality to Jade

fy17ohhh committed:
Fix format

fy17ohhh committed:
Fix format error in messages

fy17ohhh committed:
Set Title for GUI

fy17ohhh committed:
Handle the condition where start date is later than end date

fy17ohhh committed:
Handle non-existent dates

fy17ohhh committed:
Fix minor format error

fy17ohhh committed:
Merge branch 'branch-A-MoreErrorHandling'
Improve the code to handle invalid date ranges and non-existent dates

fy17ohhh committed:
Add more tests for different tasks

fy17ohhh committed:
Add JUnit tests for Deadline and Parser

fy17ohhh committed:
Add more automated tests

fy17ohhh committed:
Bug fix: set locale for JUnit tests

fy17ohhh committed:
Use ChatGPT to add and improve tests

fy17ohhh committed:
Add documentation using ChatGPT

fy17ohhh committed:
Merge branch 'branch-A-AiAssisted'
Enhance the code using ChatGPT 3.5

fy17ohhh committed:
Add load error message printing for tests

fy17ohhh committed:
Print IO Exception msg

fy17ohhh committed:
Add Bye Command in Help Page

fy17ohhh committed:
Add a brief User Guide and a representative screenshot

fy17ohhh committed:
Adjust Ui size in README

fy17ohhh committed:
Fix minor format errors in README

fy17ohhh committed:
Bug Fix: Set up Jekyll to make pages render correctly

fy17ohhh committed:
Update _config.yml

fy17ohhh committed:
Update _config.yml

fy17ohhh committed:
Update index.markdown

fy17ohhh committed:
Bug Fix: Todo command to add task properly

fy17ohhh committed:
Update Ui and delete logo

fy17ohhh committed:
Fix minor typo and update Ui.png

fy17ohhh committed:
Fix typo in AddCommand

fy17ohhh committed:
Bug fix: OS-independent loading and storage
